ast/parser.go

```diff
@@ -556,13 +557,52 @@ func (p *Parser) parseRules() []*Rule {
 		return []*Rule{&rule}
 	}
 
-	if p.s.tok == tokens.LBrace {
+	hasIf := false
+	if p.s.tok == tokens.If {
+		hasIf = true
+	}
+
+	if hasIf && !usesContains && rule.Head.Key != nil && rule.Head.Value == nil {
+		p.illegal("invalid for partial set rule %s (use `contains`)", rule.Head.Name)
```

Comment:
We disallow 
```
p[x] if { ... }
```
because we want `if` to disambiguate partial sets from complete rules for ref-heads later.